File: 2003/IntroToVBNETPart2/Tutorial/Count.cs
using System ;

public class Count
{
	const int EOF = -1 ;

	public static void Main ()
	{
		int i, nwhite, nother ;
		char c ;
		int [] ndigit = new int [10] ;

		nwhite = nother = 0 ;
		for ( i = 0 ; i < 10 ; ++i )
			ndigit[i] = 0 ;

		while ( ( i = Console.Read () ) != EOF )
		{
			c = (char) i ;
			if ( c >= '0' && ( c <= '9' ) )
				++ndigit [c - '0' ] ;
			else if ( c == ' ' || c == '\n' || c == '\t' )
				++nwhite ;
			else
				++nother ;
		}
		Console.Write ( "digits =" ) ;
		for ( i = 0 ; i < 10 ; ++i )
			Console.Write ( " {0}", ndigit[i] ) ;
		Console.WriteLine ( "\nwhite space = {0}, other = {1}", 
			nwhite, nother ) ;
	}
}

File: 2003/IntroToVBNETPart2/Tutorial/Count.vb
Imports System
Imports Microsoft.VisualBasic 

Public Class Count

	Const EOF As Integer = -1

	Shared Sub Main ()
		Dim i, nwhite, nother As Integer
		Dim c As Char
		Dim ndigit(10) As Integer

		nwhite = 0 : nother = 0
		For i = 0 To 10
			ndigit(i) = 0
		Next i

		While True
			i = Console.Read ()
			If i = EOF Then Exit While
			c = Chr ( i )
			If c >= "0"C And c <= "9"C Then
				ndigit( Asc(c) - Asc("0"C) ) += 1
			Else if c = " "C Or c = vbLf Or c = vbTab Then
				nwhite += 1
			Else
				nother += 1
			End If

		End While
		Console.Write ( "digits =" ) 
		For i = 0 To 9
			Console.Write ( " {0}", ndigit(i) ) 
		Next i
		Console.WriteLine ( vbCrLf & "white space = {0}, other = {1}", _
			nwhite, nother ) 
	End Sub
End Class